### DISCUSSION TITLE
AWS Sagemaker Classic Studio - "//data/VV : No such file or directory"

### DISCUSSION THREAD

**Discussion by mthsdiniz-usp** [isce-framework/dolphin#509] [Q&A]
Hello all,

I'm trying to run the [basic walkthrough ](https://dolphin-insar.readthedocs.io/en/latest/notebooks/walkthrough-basic/) in AWS Sagemaker Classic Studio but i'm facing some issues while trying to execute the dolphin "run" command.

When checking the files with gdalinfo or even importing gdal from osgeo, i can see the metadata for the files and the subdatasets associated with it:

![image](https://github.com/user-attachments/assets/795c415c-9a5d-4709-ae65-348c242d77b2)
![image](https://github.com/user-attachments/assets/22c9b153-eb0e-41ab-b492-e841cbcd5c38)

When executing the dolphin "run", i get a RuntimeError stating that gdal was not able to find the subdataset "//data/VV".

![image](https://github.com/user-attachments/assets/41e31ba8-7b76-47ec-9b04-a99e4f3d0f6e)

Did anyone had this error before or anything related to AWS Sagemaker Classic Studio?

Thanks!

** Here is how the environment is setup:

![image](https://github.com/user-attachments/assets/28a8da25-26e6-4553-8c64-86fcba9e7240)

**Comment by scottstanie** [comment]
Hmm I actually hadn't heard of Sagemaker until now. How does the environment installation work?

One likely reason is from GDAL changing how the driver installation works when you install through conda. To read those SLC files, you also need https://anaconda.org/conda-forge/libgdal-netcdf (and possibly https://anaconda.org/conda-forge/libgdal-hdf5 ) to be installed

**Comment by mthsdiniz-usp** [comment]
> Hmm I actually hadn't heard of Sagemaker until now. How does the environment installation work?
> 
> One likely reason is from GDAL changing how the driver installation works when you install through conda. To read those SLC files, you also need https://anaconda.org/conda-forge/libgdal-netcdf (and possibly https://anaconda.org/conda-forge/libgdal-hdf5 ) to be installed

Sagemaker is a Jupyter Lab environment which all dependencies are setup using a Dockerfile. Right now i'm using a python-gdal-hf5 (https://hub.docker.com/r/donmichael/python-gdal-hdf5) as my base and installing the remainder python requirements using pip.

I think all the requirements might be fine since i can open an read data when importing gdal from osgeo or using gdalinfo. I'm gonna try using a conda base image to setup the environment with the dependencies you mentioned and try again.

**Reply by scottstanie** [reply]
So i see that it recognizes with `gdalinfo hdf5:/path....://data/VV`. what about with `gdalinfo netcdf:/path....:/data/VV` ? that it is a `RuntimeError` is telling me it's something with GDAL, which I know has become harder to set up recently

**Reply by mthsdiniz-usp** [reply]
> So i see that it recognizes with `gdalinfo hdf5:/path....://data/VV`. what about with `gdalinfo netcdf:/path....:/data/VV` ? that it is a `RuntimeError` is telling me it's something with GDAL, which I know has become harder to set up recently

I think that was the issue, i had hdf5 format enabled in GDAL but NetCDF was not. In any case i shifted for an Ubuntu docker image and i'm setting up miniconda and all dependencies manually. 

As soon as i'm finish i can just upload here so other users can make use of it :)

I'm running into another problem now but i didn't search yet for solutions. Adding it here just in case

![image](https://github.com/user-attachments/assets/6d2865a6-3d64-4ef2-ba55-447ee6633587)

**Reply by scottstanie** [reply]
hm that's one I haven't seen... if that file does exist, but gdal can't recognize the .tif, then that would be very weird to me. just checking- this new one wasn't from trying to run some partially-completed run after a restart?

**Reply by mthsdiniz-usp** [reply]
Hey @scottstanie, you were right, i deleted the folder and fixed that problem. 

I had to make some changes in my docker image and on the notebook in order to make it recognize pyproj...

Right now i'm using an Ubuntu base image, installing miniconda and all dependencies using conda. Whenever executing the code, i also need to import pyproj and set an environment variable pointing to the directory it's database is located with the following code:

```## Setup pyproj directory using environment variables 
import pyproj
os.environ['PROJ_LIB'] = pyproj.datadir.get_data_dir()
```
After implementing this changes the process runs for quite a good time (i'm processing almost 4 years worth of data) but after stitching interferograms for the whole timeseries it throws the following error:

![image](https://github.com/user-attachments/assets/062c08be-24fe-407f-9bd1-973d359b1497)

I tried looking at different version of pyproj but none of them seems to have a LAYOUT.VERSION.MINOR >= 3

![image](https://github.com/user-attachments/assets/74d87430-f217-40e4-bba3-f973ca6b8eb1)


Have you seen this problem before?

**Reply by scottstanie** [reply]
hmmmm I think once I've seen the `PROJ: internal_proj_create_from_database` error, but it was from me installing then reinstalling stuff in the same conda environment so that the gdal, proj, and rasterio expectations were mismatched... it's an unfortunately annoying environment bug to figure out :\

**Reply by mthsdiniz-usp** [reply]
@scottstanie you were right! I was installing pyproj using conda in my docker image and later on installing Dolphin, which caused a mismatch between the conda version and the Dolphin one. Once i figured it out, it was just a matter of setting the right path into the PROJ_LIB env variable and it worked.

I have some general questions around Insar, MintPy and Dolphin, especially around reference date. Can i ask them here?

**Reply by scottstanie** [reply]
If you mean "here" in the discussions, go right ahead! I might open a new Discussions thread so that others who have similar mintpy-related questions have an easier time searching/finding your answers
